proposals/accepted/047-standard-library-reform.rst

```diff
@@ -0,0 +1,559 @@
+
+.. _`GHC issue #21738`: https://gitlab.haskell.org/ghc/ghc/-/issues/21738
+.. _`GHC proposal #529`: https://github.com/ghc-proposals/ghc-proposals/pull/529
+
+Bonus **Step 3B**: Rethinking Windows
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Right now, ``base`` relies on MinGW's and Windows's `libc` compat layer to approximate traditional Unix functionality.
+The ``unix`` and ``Win32`` layers than expose additional platform-specific functionality.
+
+Quite arguably, this is the wrong way of going about IO.
+
+- It would be nice to make MinGW optional and support Windows more directly/natively.
+  This is what Rust does.
+  LLVM has made doing so (e.g. without relying on proprietary tools exclusively) much easier in recent years.
+  As Ben Gamari and others can attest, the state of Windows support in GNU tools is not good.
+
+- It would be nice to not limit ourselves to a lowest-common-denominator of ``libc``-esque functionality as our starting point.
+  Windows and Linux have added all sorts of more modern functionality in recent years that often is (a) similar, and (b) represents better ways to do existing operations, e.g. avoiding around restrictions on character sets, file path length, etc.
+
+From this perspective we should invert the dependencies:
+``unix`` and ``Win32`` should be below, binding Unix and Windows APIs *as they are*,
+and then *above* that is a compatibility layer creating portable interfaces with the latest best practice *without* the burden of libc tradition.
+
+This sort of reshuffle is a continuation of the project of rationalizing dependencies and a natural extension of **Step 2A**.
+
+Timeline
+--------
+
+The project is designed to proceed in parallel to minimize risk, in addition to being incremental.
+Steps 1a and 1b are independent, and steps 2a and 2b are likewise independent.
+
+In past discussion, consensus around a plan from **Step 1A** was emphasized as a blocker --- if we didn't know what sort of standard libraries we wanted to end up with, we shouldn't proceed.
+In the author's opinion this is misguided.
+The actual stumbling point is not disagreements about where we want to end up, but maintaining progress on something which is not incredibly hard, but has many steps and ushers in most of the benefit over the long term.
+(For example, many users of GHC are behind the latest version, these reforms only benefit them going forward after they have caught up to the last unaffected release.)
+
+As such, the most crucial step is considered to be **Step 1B**.
+After that, we know the basic concept for sure works.
+And indeed it is possible to start steps 2a and 2b before there is a complete **Step 1A** plan.
+
+It may well additionally make sense to preliminarily accept *just* **Step 1B**, and then go back and refine this proposal's Timeline and Budget sections with the information we've learned from **Step 1B**.
+
+Budget
+------
+
+**Step 1A** costs
```

Comment:
Once again, named references instead of numeric references would improve readability